contractsV5/KyberNetwork.sol

```diff
@@ -58,23 +55,14 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
     // the new trade with hint
     function tradeWithHintAndFee(address payable trader, IERC20 src, uint srcAmount, IERC20 dest, address payable destAddress,
         uint maxDestAmount, uint minConversionRate, address payable platformWallet, uint platformFeeBps, bytes calldata hint)
-        external payable 
+        external payable
         returns(uint destAmount)
-    {    
+    {
         TradeData memory tradeData;
-
-        tradeData.input.trader = trader;
-        tradeData.input.src = src;
-        tradeData.input.srcAmount = srcAmount;
-        tradeData.input.dest = dest;
-        tradeData.input.destAddress = destAddress;
-        tradeData.input.maxDestAmount = maxDestAmount;
-        tradeData.input.minConversionRate = minConversionRate;
-        tradeData.input.platformWallet = platformWallet;
-        tradeData.input.platformFeeBps = platformFeeBps;
-
+        initTradeData(trader, src, srcAmount, dest, destAddress, maxDestAmount, minConversionRate, platformWallet, platformFeeBps, tradeData);
```

Comment:
not sure why is this better?
and why here one liner and next one is one below other?
anyway, if keeping this long function call
please use safe call convention . with parameter names
i.e.

callFuction(paramName: input, nextParamName: input2 etc...

Reply:
Will swtich to the one below the other.

Reply:
1) Abstraction, since it's being used by the trade and getExpectedRate functions. 
2) Just need to change the underlying the function, and it ensures all the relevant inputs are initialised. 
3) Run the risk of missing an initialisation of a param, or wrong initialisation.

Agree with the safe call convention if we're deciding to stick to it. I've reverted to as is for now.